diff --git i/examples/boot/bootloader/rp/src/main.rs w/examples/boot/bootloader/rp/src/main.rs
index 25b1657b8..4c3bfbaa3 100644
--- i/examples/boot/bootloader/rp/src/main.rs
+++ w/examples/boot/bootloader/rp/src/main.rs
@@ -24,7 +24,8 @@ fn main() -> ! {
     }
     */
 
-    let flash = WatchdogFlash::<FLASH_SIZE>::start(p.FLASH, p.WATCHDOG, Duration::from_secs(8));
+    use embassy_rp::flash::{Flash, Blocking};
+    let flash = Flash::<_, Blocking, FLASH_SIZE>::new_blocking(p.FLASH);
     let flash = Mutex::new(RefCell::new(flash));
 
     let config = BootLoaderConfig::from_linkerfile_blocking(&flash, &flash, &flash);
